wasmtime/_memory.py

```diff
@@ -62,6 +62,73 @@ def data_ptr(self, store: Storelike) -> "ctypes._Pointer[c_ubyte]":
         """
         return ffi.wasmtime_memory_data(store._context, byref(self._memory))
 
+    def set_store(self, store: Storelike):
+        """
+        must be called to set the store for highlevel access of memory
+        """
+        self._store=store
+
+    def __getitem__(self, key: int):
+        """
+        provide memory[offset] or memory[start:stop]
+        """
+        if self._store is None:
+            raise RuntimeError("you must call `set_store()` before using highlevel access")
+        data_ptr = self.data_ptr(self._store)
+        size = self.data_len(self._store)
+        if isinstance(key, int):
+            if key>=size:
+                raise IndexError("out of memory size")
+            return data_ptr[key]
+        if not isinstance(key, slice):
+            raise TypeError("key can either be integer index or slice")
+        
+        start, stop, step = key.indices(size)
+        if step!=1:
+            raise ValueError("slice with step is not supported")
+
+        val_size = stop - start
+        value = bytearray(val_size)
+
+        if stop>size:
+            raise IndexError("out of memory size")
+        dst_ptr = (ctypes.c_ubyte * val_size).from_buffer(value)
+        src_ptr = ctypes.addressof((ctypes.c_ubyte*val_size).from_address(ctypes.addressof(data_ptr.contents)+start))
+        ctypes.memmove(dst_ptr, src_ptr, val_size)
```

Comment:
I think this may be able to be simplified a bit. There's a `to_bytes` helper in the `_ffi.py` file which I believe does roughly what this is doing. Could that helper function be used here, or perhaps the helper refactored a bit to be called with the pointer/len available on-hand here?

Reply:
yes, the logic of `to_bytes.()` worked fine, and it was fast. we just need away to pass custom offset and custom length

```python
def to_bytes(vec: wasm_byte_vec_t, offset=0, size=None) -> bytearray:
    if size is None: size=vec.size
    ty = c_uint8 * size
    return bytearray(ty.from_address(addressof(vec.data.contents)+offset))
```

Reply:
Something like that is fine, yeah, or alternatively the logic inside of `to_bytes` could be split out to a separate function which takes a pointer/len argument and `to_bytes`, as-is right now, forwards to that.